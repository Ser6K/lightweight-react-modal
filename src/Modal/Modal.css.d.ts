declare namespace ModalCssNamespace {
  export interface IModalCss {
    modal: string;
    wrapper: string;
  }
}

declare const ModalCssModule: ModalCssNamespace.IModalCss & {
  /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
  locals: ModalCssNamespace.IModalCss;
};

export = ModalCssModule;
