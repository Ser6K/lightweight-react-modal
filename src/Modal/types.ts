import { ClassNameType, ReactNodeType } from 'src/types'

export interface ModalTypes {
  onClose?: () => void
  fluid?: boolean
  closable?: boolean
  maxHeight?: number
  minHeight?: number
  maxWidth?: number
  minWidth?: number
  children: ReactNodeType
  customClassNames?: {
    wrapper: ClassNameType
    modal: ClassNameType
    closeBtn: ClassNameType
    overlay: ClassNameType
  }
  closeButtonIcon?: ReactNodeType
  [propName: string]: any
}
