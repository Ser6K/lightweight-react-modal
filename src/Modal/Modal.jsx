import React, {
    useState,
    useCallback,
    useEffect,
    useRef,
} from 'react';
import { createPortal } from 'react-dom';
import classNames from 'classnames';
import { createUseStyles } from 'react-jss';
import PropTypes from 'prop-types';
import {
    getHeader,
    getFooter,
    getContent,
    handleEscPress,
} from '../utils/utils';
import { addModal, removeModal } from '../utils/register';
import CloseButton from './components/CloseButton/CloseButton';
import Overlay from './components/Overlay/Overlay';
import styles from './Modal.styles';

const Modal = ({
    onClose,
    fluid,
    closable,
    maxHeight,
    minHeight,
    maxWidth,
    minWidth,
    children,
    customClassNames,
}) => {
    const header = useCallback(getHeader(children), [children]);
    const footer = useCallback(getFooter(children), [children]);
    const content = useCallback(getContent(children), [children]);
    const modalRef = useRef();

    const classes = createUseStyles(styles)({
        fluid, maxHeight, minHeight, maxWidth, minWidth,
    });

    useEffect(() => {
        addModal(modalRef);
    }, [modalRef]);

    const escPressed = handleEscPress(closable, modalRef);
    const closeModal = useCallback(() => {
        if (!closable || !onClose) {
            return;
        }
        removeModal(modalRef);
        onClose();
    }, [closable, onClose, modalRef]);

    useEffect(() => {
        if (escPressed) {
            closeModal();
        }
    }, [escPressed]);

    return createPortal(
        <div className={classNames(classes.wrapper, customClassNames.wrapper)}>
            <div
                ref={modalRef}
                className={classNames(classes.modal, customClassNames.modal)}
            >
                {closable && (
                    <CloseButton
                        className={customClassNames.closeButton}
                        onClick={closeModal}
                    />
                )}
                {header && header}
                {content && content}
                {footer && footer}
            </div>
            <Overlay
                onClick={closeModal}
                className={customClassNames.overlay}
            />
        </div>, document.body,
    );
};

Modal.propTypes = {
    onClose: PropTypes.func,
    fluid: PropTypes.bool,
    closable: PropTypes.bool,
    maxHeight: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string,
    ]),
    minHeight: PropTypes.number,
    maxWidth: PropTypes.number,
    minWidth: PropTypes.number,
    children: PropTypes.node.isRequired,
    customClassNames: PropTypes.shape({
        wrapper: PropTypes.string,
        modal: PropTypes.string,
        closeBtn: PropTypes.string,
        overlay: PropTypes.string,
    }),
};

Modal.defaultProps = {
    onClose: null,
    fluid: false,
    closable: true,
    maxHeight: 500,
    minHeight: 100,
    maxWidth: 500,
    minWidth: 200,
    customClassNames: {
        wrapper: null,
        modal: null,
        closeButton: null,
        overlay: null,
    },
};

Modal.displayName = 'Modal';
export default React.memo(Modal);
