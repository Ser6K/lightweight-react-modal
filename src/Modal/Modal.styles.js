import styled from 'styled-components';

export const Wrapper = styled.div`
    position: fixed;
    z-index: 99998;
    top: 0;
    bottom: 0;
    right: 0;
    left: 0;
    display: flex;
    align-items: center;
    justify-content: center;
`;

export const ModalWrapper = styled.div`
    position: relative;
    z-index: 99999;
    background: white;
    box-sizing: border-box;
    border: 1px solid grey;
    border-radius: 3px;
    height: 90%;
    width: 90%;
    max-width: ${props => !props.fluid ? `${props.maxWidth}px` : ''};
    min-width: ${props => !props.fluid ? `${props.minWidth}px` : ''};
    max-height: ${props => !props.fluid ? `${props.maxHeight}px` : ''};
    min-height: ${props => !props.fluid ? `${props.minHeight}px` : ''};
    display: flex;
    flex-direction: column;
    flex-wrap: nowrap;
`;
