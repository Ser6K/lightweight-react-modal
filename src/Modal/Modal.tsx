import React, { useCallback, useEffect, useRef } from 'react'
import { createPortal } from 'react-dom'
import { getHeader, getFooter, getContent, handleEscPress, classNames, addModal, removeModal } from 'src/utils'
import CloseButton from './components/CloseButton/CloseButton'
import Overlay from './components/Overlay/Overlay'
import styles from './Modal.css'

interface ModalProps {
  onClose?: () => void
  fluid?: boolean
  closable?: boolean
  maxHeight?: number
  minHeight?: number
  maxWidth?: number
  minWidth?: number
  children: React.ReactNode
  customClassNames?: {
    wrapper: string | null
    modal: string | null
    closeBtn: string | null
    overlay: string | null
  }
  closeButtonIcon?: React.ReactNode
  [propName: string]: any
}

const Modal: React.FC<ModalProps> = ({
  onClose,
  fluid = false,
  closable = true,
  maxHeight = 500,
  minHeight = 100,
  maxWidth = 500,
  minWidth = 200,
  children = null,
  customClassNames = {
    wrapper: null,
    modal: null,
    closeBtn: null,
    overlay: null,
  },
  closeButtonIcon = null,
  ...props
}) => {
  const header = useCallback(getHeader(children), [children])
  const footer = useCallback(getFooter(children), [children])
  const content = useCallback(getContent(children), [children])
  const modalRef = useRef(null)

  useEffect(() => {
    addModal(modalRef)
  }, [modalRef])

  const escPressed = handleEscPress(closable, modalRef)

  const closeModal = useCallback(() => {
    if (!closable || !onClose) {
      return
    }
    removeModal(modalRef)
    onClose()
  }, [closable, onClose, modalRef])

  useEffect(() => {
    if (escPressed) {
      closeModal()
    }
  }, [escPressed])

  return createPortal(
    <div {...props} className={classNames(styles.wrapper, customClassNames.wrapper)}>
      <div
        ref={modalRef}
        className={classNames(styles.modal, customClassNames.modal)}
        style={{
          maxWidth: `${!fluid ? `${maxWidth}px` : ''}`,
          minWidth: `${!fluid ? `${minWidth}px` : ''}`,
          maxHeight: `${!fluid ? `${maxHeight}px` : ''}`,
          minHeight: `${!fluid ? `${minHeight}px` : ''}`,
        }}
      >
        {closable && (
          <CloseButton onClick={closeModal} icon={closeButtonIcon} {...(customClassNames.closeBtn != null ? { className: customClassNames.closeBtn } : {})} />
        )}
        {header && header}
        {content && content}
        {footer && footer}
      </div>
      <Overlay onClick={closeModal} className={customClassNames.overlay} />
    </div>,
    document.body
  )
}

export default React.memo(Modal)
