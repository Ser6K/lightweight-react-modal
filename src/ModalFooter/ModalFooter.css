.footer {
    flex-shrink: 0;
    border-top: 1px solid grey;
    padding: 15px 20px;
    box-sizing: border-box;
}
