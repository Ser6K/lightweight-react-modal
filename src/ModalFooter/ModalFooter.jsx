import React from 'react';
import PropTypes from 'prop-types';
import { FooterDisplayName } from '../utils/utils';

const ModalFooter = ({ children }) => children;

ModalFooter.displayName = FooterDisplayName;

ModalFooter.propTypes = {
    children: PropTypes.node.isRequired,
};

export default ModalFooter;
