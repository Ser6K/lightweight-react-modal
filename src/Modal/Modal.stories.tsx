import React, { useState } from 'react'
import ModalHeader from '../ModalHeader/ModalHeader'
import ModalFooter from '../ModalFooter/ModalFooter'
import ModalContent from '../ModalContent/ModalContent'
import Modal from './Modal'

export default { title: 'Modal' }

export const defaultExample = () => (
  <Modal closable={false} style={{ textAlign: 'center' }}>
    <ModalHeader>
      <h1>Modal Header</h1>
    </ModalHeader>
    <ModalContent>
      Sportsman delighted improving dashwoods gay instantly happiness six. Ham now amounted absolute not mistaken way pleasant whatever. At an these still no
      dried folly stood thing. Rapid it on hours hills it seven years. If polite he active county in spirit an. Mrs ham intention promotion engrossed assurance
      defective. Confined so graceful building opinions whatever trifling in. Insisted out differed ham man endeavor expenses. At on he total their he songs.
      Related compact effects is on settled do. Sportsman delighted improving dashwoods gay instantly happiness six. Ham now amounted absolute not mistaken way
      pleasant whatever. At an these still no dried folly stood thing. Rapid it on hours hills it seven years. If polite he active county in spirit an. Mrs ham
      intention promotion engrossed assurance defective. Confined so graceful building opinions whatever trifling in. Insisted out differed ham man endeavor
      expenses. At on he total their he songs. Related compact effects is on settled do. Sportsman delighted improving dashwoods gay instantly happiness six.
      Ham now amounted absolute not mistaken way pleasant whatever. At an these still no dried folly stood thing. Rapid it on hours hills it seven years. If
      polite he active county in spirit an. Mrs ham intention promotion engrossed assurance defective. Confined so graceful building opinions whatever trifling
      in. Insisted out differed ham man endeavor expenses. At on he total their he songs. Related compact effects is on settled do. Sportsman delighted
      improving dashwoods gay instantly happiness six. Ham now amounted absolute not mistaken way pleasant whatever. At an these still no dried folly stood
      thing. Rapid it on hours hills it seven years. If polite he active county in spirit an. Mrs ham intention promotion engrossed assurance defective.
      Confined so graceful building opinions whatever trifling in. Insisted out differed ham man endeavor expenses. At on he total their he songs. Related
      compact effects is on settled do.
    </ModalContent>
    <ModalFooter>
      <h1>Modal Footer</h1>
    </ModalFooter>
  </Modal>
)

export const fluidExample = () => (
  <Modal closable={false} fluid style={{ textAlign: 'center' }}>
    <ModalHeader>
      <h1>Modal Header</h1>
    </ModalHeader>
    <ModalContent>
      Sportsman delighted improving dashwoods gay instantly happiness six. Ham now amounted absolute not mistaken way pleasant whatever. At an these still no
      dried folly stood thing. Rapid it on hours hills it seven years. If polite he active county in spirit an. Mrs ham intention promotion engrossed assurance
      defective. Confined so graceful building opinions whatever trifling in. Insisted out differed ham man endeavor expenses. At on he total their he songs.
      Related compact effects is on settled do. Sportsman delighted improving dashwoods gay instantly happiness six. Ham now amounted absolute not mistaken way
      pleasant whatever. At an these still no dried folly stood thing. Rapid it on hours hills it seven years. If polite he active county in spirit an. Mrs ham
      intention promotion engrossed assurance defective. Confined so graceful building opinions whatever trifling in. Insisted out differed ham man endeavor
      expenses. At on he total their he songs. Related compact effects is on settled do. Sportsman delighted improving dashwoods gay instantly happiness six.
      Ham now amounted absolute not mistaken way pleasant whatever. At an these still no dried folly stood thing. Rapid it on hours hills it seven years. If
      polite he active county in spirit an. Mrs ham intention promotion engrossed assurance defective. Confined so graceful building opinions whatever trifling
      in. Insisted out differed ham man endeavor expenses. At on he total their he songs. Related compact effects is on settled do. Sportsman delighted
      improving dashwoods gay instantly happiness six. Ham now amounted absolute not mistaken way pleasant whatever. At an these still no dried folly stood
      thing. Rapid it on hours hills it seven years. If polite he active county in spirit an. Mrs ham intention promotion engrossed assurance defective.
      Confined so graceful building opinions whatever trifling in. Insisted out differed ham man endeavor expenses. At on he total their he songs. Related
      compact effects is on settled do.
    </ModalContent>
    <ModalFooter>
      <h1>Modal Footer</h1>
    </ModalFooter>
  </Modal>
)

export const closableModalExample = () => {
  const [open, setOpen] = useState(true)

  return (
    <>
      <button
        onClick={() => {
          setOpen(true)
        }}
        type="button"
      >
        Open Modal
      </button>
      {open ? (
        <Modal
          style={{ textAlign: 'center' }}
          onClose={() => {
            setOpen(false)
          }}
        >
          <ModalHeader>
            <h1>Modal Header</h1>
          </ModalHeader>
          <ModalContent>
            Sportsman delighted improving dashwoods gay instantly happiness six. Ham now amounted absolute not mistaken way pleasant whatever. At an these still
            no dried folly stood thing. Rapid it on hours hills it seven years. If polite he active county in spirit an. Mrs ham intention promotion engrossed
            assurance defective. Confined so graceful building opinions whatever trifling in. Insisted out differed ham man endeavor expenses. At on he total
            their he songs. Related compact effects is on settled do. Sportsman delighted improving dashwoods gay instantly happiness six. Ham now amounted
            absolute not mistaken way pleasant whatever. At an these still no dried folly stood thing. Rapid it on hours hills it seven years. If polite he
            active county in spirit an. Mrs ham intention promotion engrossed assurance defective. Confined so graceful building opinions whatever trifling in.
            Insisted out differed ham man endeavor expenses. At on he total their he songs. Related compact effects is on settled do. Sportsman delighted
            improving dashwoods gay instantly happiness six. Ham now amounted absolute not mistaken way pleasant whatever. At an these still no dried folly
            stood thing. Rapid it on hours hills it seven years. If polite he active county in spirit an. Mrs ham intention promotion engrossed assurance
            defective. Confined so graceful building opinions whatever trifling in. Insisted out differed ham man endeavor expenses. At on he total their he
            songs. Related compact effects is on settled do. Sportsman delighted improving dashwoods gay instantly happiness six. Ham now amounted absolute not
            mistaken way pleasant whatever. At an these still no dried folly stood thing. Rapid it on hours hills it seven years. If polite he active county in
            spirit an. Mrs ham intention promotion engrossed assurance defective. Confined so graceful building opinions whatever trifling in. Insisted out
            differed ham man endeavor expenses. At on he total their he songs. Related compact effects is on settled do.
          </ModalContent>
          <ModalFooter>
            <h1>Modal Footer</h1>
          </ModalFooter>
        </Modal>
      ) : null}
    </>
  )
}

export const multiModalExample = () => {
  const [open, setOpen] = useState(true)
  const [secondOpen, setSecondOpen] = useState(false)

  return (
    <>
      <button
        onClick={() => {
          setOpen(true)
        }}
        type="button"
      >
        Open Modal
      </button>
      {open ? (
        <Modal
          style={{ textAlign: 'center' }}
          onClose={() => {
            setOpen(false)
          }}
        >
          <ModalHeader>
            <h1>Modal Header</h1>
          </ModalHeader>
          <ModalContent>
            <p>Click to Open second modal!</p>
            <button
              onClick={() => {
                setSecondOpen(true)
              }}
              type="button"
            >
              Open Second Modal
            </button>
          </ModalContent>
          <ModalFooter>
            <h1>Modal Footer</h1>
          </ModalFooter>
        </Modal>
      ) : null}
      {secondOpen ? (
        <Modal
          style={{ textAlign: 'center' }}
          onClose={() => {
            setSecondOpen(false)
          }}
        >
          <ModalHeader>
            <h1>Second Modal Header</h1>
          </ModalHeader>
          <ModalContent>
            Sportsman delighted improving dashwoods gay instantly happiness six. Ham now amounted absolute not mistaken way pleasant whatever. At an these still
            no dried folly stood thing. Rapid it on hours hills it seven years. If polite he active county in spirit an. Mrs ham intention promotion engrossed
            assurance defective. Confined so graceful building opinions whatever trifling in. Insisted out differed ham man endeavor expenses. At on he total
            their he songs. Related compact effects is on settled do.
          </ModalContent>
          <ModalFooter>
            <h1>Second Modal Footer</h1>
          </ModalFooter>
        </Modal>
      ) : null}
    </>
  )
}
