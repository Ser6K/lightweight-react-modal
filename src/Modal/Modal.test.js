import React, { useState } from 'react';
import ReactDom from 'react-dom';
import { shallow, mount } from 'enzyme';
import Modal from './Modal';
import * as ModalRegister from '../utils/register';
import ModalHeader from '../ModalHeader/ModalHeader';
import ModalFooter from '../ModalFooter/ModalFooter';

describe('Modal tests', () => {
    let WrappedModal;

    beforeEach(() => {
        WrappedModal = () => {
            const [open, setOpen] = useState(true);

            return open && (
                <Modal onClose={() => setOpen(false)}>
                    Modal
                </Modal>
            );
        };
    });

    test('it can be mounted', () => {
        const modal = mount(<Modal>Modal</Modal>);
        expect(modal.children()).toHaveLength(1);
    });

    test('it matches snapshot', () => {
        const modal = mount(<Modal><div>dialog</div><div>contetn</div></Modal>);
        expect(modal).toMatchSnapshot();
    });

    test('it can register modal reference on mount', () => {
        const modalRegisterSpy = jest.spyOn(ModalRegister, 'addModal');
        mount(<Modal>Modal</Modal>);

        expect(modalRegisterSpy).toHaveBeenCalledTimes(1);
    });

    test('it mounts with create portal', () => {
        const portalSpy = jest.spyOn(ReactDom, 'createPortal');
        shallow(<Modal>Modal</Modal>);

        expect(portalSpy).toHaveBeenCalledTimes(1);
    });

    test('it can be mounted with header', () => {
        const modal = mount(<Modal><ModalHeader>Header</ModalHeader></Modal>);
        expect(modal.find('ModalHeader').text()).toBe('Header');
    });

    test('it doesn\'t mount header if it\'s empty', () => {
        const modal = mount(<Modal><ModalHeader /></Modal>);
        expect(modal.find('ModalHeader')).toEqual({});
    });

    test('it can be mounted with footer', () => {
        const modal = mount(<Modal><ModalFooter>Footer</ModalFooter></Modal>);
        expect(modal.find('ModalFooter').text()).toBe('Footer');
    });

    test('it doesn\'t mount footer if it\'s empty', () => {
        const modal = mount(<Modal><ModalFooter /></Modal>);
        expect(modal.find('ModalFooter')).toEqual({});
    });

    test('fluid modal matches snapshot', () => {
        const modal = mount(<Modal fluid>Modal</Modal>);
        expect(modal).toMatchSnapshot();
    });

    test('not closable modal matches snapshot', () => {
        const modal = mount(<Modal closable={false}>Modal</Modal>);
        expect(modal).toMatchSnapshot();
    });

    test('it closes with close btn click', () => {
        const modal = mount(<WrappedModal />);
        const closeBtn = modal.find('closeButton');
        closeBtn.simulate('click');
        expect(modal.children()).toHaveLength(0);
    });

    test('it closes with overlay click', () => {
        const modal = mount(<WrappedModal />);
        const modalWrapper = modal.find('div[onClick]');
        modalWrapper.simulate('click');
        expect(modal.children()).toHaveLength(0);
    });

    test('it closes with close btn click', () => {
        const modal = mount(<WrappedModal />);
        const modalRegisterSpy = jest.spyOn(ModalRegister, 'removeModal');
        const closeBtn = modal.find('closeButton');
        closeBtn.simulate('click');
        expect(modalRegisterSpy).toHaveBeenCalledTimes(1);
    });

    test('it can unsubscribe reference on unmount with overlay click', () => {
        const modal = mount(<WrappedModal />);
        const modalRegisterSpy = jest.spyOn(ModalRegister, 'removeModal');
        const modalWrapper = modal.find('div[onClick]');
        modalWrapper.simulate('click');
        expect(modalRegisterSpy).toHaveBeenCalledTimes(1);
    });
});
