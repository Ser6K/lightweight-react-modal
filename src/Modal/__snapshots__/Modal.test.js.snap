// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal tests fluid modal matches snapshot 1`] = `
<Modal
  fluid={true}
>
  <Portal
    containerInfo={
      <body>
        <div
          class="wrapper-0-2-1"
        >
          <div
            class="modal-0-2-2 modal-d0-0-2-8"
          >
            <button
              class="close-0-2-4"
              role="button"
            >
              x
            </button>
            <div
              class="content-0-2-5"
            >
              Modal
            </div>
          </div>
        </div>
        <div
          class="overlay-0-2-7"
        />
        <div
          class="wrapper-0-2-9"
        >
          <div
            class="modal-0-2-10 modal-d0-0-2-16"
          >
            <button
              class="close-0-2-12"
              role="button"
            >
              x
            </button>
            <div
              class="content-0-2-13"
            >
              <div>
                dialog
              </div>
              <div>
                contetn
              </div>
            </div>
          </div>
        </div>
        <div
          class="overlay-0-2-15"
        />
        <div
          class="wrapper-0-2-17"
        >
          <div
            class="modal-0-2-18 modal-d0-0-2-24"
          >
            <button
              class="close-0-2-20"
              role="button"
            >
              x
            </button>
            <div
              class="content-0-2-21"
            >
              Modal
            </div>
          </div>
        </div>
        <div
          class="overlay-0-2-23"
        />
        <div
          class="wrapper-0-2-33"
        >
          <div
            class="modal-0-2-34 modal-d0-0-2-40"
          >
            <button
              class="close-0-2-36"
              role="button"
            >
              x
            </button>
            <div
              class="header-0-2-35"
            >
              Header
            </div>
            <div
              class="content-0-2-37"
            />
          </div>
        </div>
        <div
          class="overlay-0-2-39"
        />
        <div
          class="wrapper-0-2-41"
        >
          <div
            class="modal-0-2-42 modal-d0-0-2-48"
          >
            <button
              class="close-0-2-44"
              role="button"
            >
              x
            </button>
            <div
              class="content-0-2-45"
            />
          </div>
        </div>
        <div
          class="overlay-0-2-47"
        />
        <div
          class="wrapper-0-2-49"
        >
          <div
            class="modal-0-2-50 modal-d0-0-2-56"
          >
            <button
              class="close-0-2-52"
              role="button"
            >
              x
            </button>
            <div
              class="content-0-2-53"
            />
            <div
              class="footer-0-2-54"
            >
              Footer
            </div>
          </div>
        </div>
        <div
          class="overlay-0-2-55"
        />
        <div
          class="wrapper-0-2-57"
        >
          <div
            class="modal-0-2-58 modal-d0-0-2-64"
          >
            <button
              class="close-0-2-60"
              role="button"
            >
              x
            </button>
            <div
              class="content-0-2-61"
            />
          </div>
        </div>
        <div
          class="overlay-0-2-63"
        />
        <div
          class="wrapper-0-2-65"
        >
          <div
            class="modal-0-2-66 modal-d0-0-2-72"
          >
            <button
              class="close-0-2-68"
              role="button"
            >
              x
            </button>
            <div
              class="content-0-2-69"
            >
              Modal
            </div>
          </div>
        </div>
        <div
          class="overlay-0-2-71"
        />
      </body>
    }
  >
    <div
      className="wrapper-0-2-65"
      onClick={[Function]}
    >
      <div
        className="modal-0-2-66 modal-d0-0-2-72"
      >
        <button
          className="close-0-2-68"
          onClick={[Function]}
          role="button"
        >
          x
        </button>
        <div
          className="content-0-2-69"
        >
          Modal
        </div>
      </div>
    </div>
    <div
      className="overlay-0-2-71"
    />
  </Portal>
</Modal>
`;

exports[`Modal tests it matches snapshot 1`] = `
<Modal>
  <Portal
    containerInfo={
      <body>
        <div
          class="wrapper-0-2-1"
        >
          <div
            class="modal-0-2-2 modal-d0-0-2-8"
          >
            <button
              class="close-0-2-4"
              role="button"
            >
              x
            </button>
            <div
              class="content-0-2-5"
            >
              Modal
            </div>
          </div>
        </div>
        <div
          class="overlay-0-2-7"
        />
        <div
          class="wrapper-0-2-9"
        >
          <div
            class="modal-0-2-10 modal-d0-0-2-16"
          >
            <button
              class="close-0-2-12"
              role="button"
            >
              x
            </button>
            <div
              class="content-0-2-13"
            >
              <div>
                dialog
              </div>
              <div>
                contetn
              </div>
            </div>
          </div>
        </div>
        <div
          class="overlay-0-2-15"
        />
      </body>
    }
  >
    <div
      className="wrapper-0-2-9"
      onClick={[Function]}
    >
      <div
        className="modal-0-2-10 modal-d0-0-2-16"
      >
        <button
          className="close-0-2-12"
          onClick={[Function]}
          role="button"
        >
          x
        </button>
        <div
          className="content-0-2-13"
        >
          <div>
            dialog
          </div>
          <div>
            contetn
          </div>
        </div>
      </div>
    </div>
    <div
      className="overlay-0-2-15"
    />
  </Portal>
</Modal>
`;

exports[`Modal tests not closable modal matches snapshot 1`] = `
<Modal
  closable={false}
>
  <Portal
    containerInfo={
      <body>
        <div
          class="wrapper-0-2-1"
        >
          <div
            class="modal-0-2-2 modal-d0-0-2-8"
          >
            <button
              class="close-0-2-4"
              role="button"
            >
              x
            </button>
            <div
              class="content-0-2-5"
            >
              Modal
            </div>
          </div>
        </div>
        <div
          class="overlay-0-2-7"
        />
        <div
          class="wrapper-0-2-9"
        >
          <div
            class="modal-0-2-10 modal-d0-0-2-16"
          >
            <button
              class="close-0-2-12"
              role="button"
            >
              x
            </button>
            <div
              class="content-0-2-13"
            >
              <div>
                dialog
              </div>
              <div>
                contetn
              </div>
            </div>
          </div>
        </div>
        <div
          class="overlay-0-2-15"
        />
        <div
          class="wrapper-0-2-17"
        >
          <div
            class="modal-0-2-18 modal-d0-0-2-24"
          >
            <button
              class="close-0-2-20"
              role="button"
            >
              x
            </button>
            <div
              class="content-0-2-21"
            >
              Modal
            </div>
          </div>
        </div>
        <div
          class="overlay-0-2-23"
        />
        <div
          class="wrapper-0-2-33"
        >
          <div
            class="modal-0-2-34 modal-d0-0-2-40"
          >
            <button
              class="close-0-2-36"
              role="button"
            >
              x
            </button>
            <div
              class="header-0-2-35"
            >
              Header
            </div>
            <div
              class="content-0-2-37"
            />
          </div>
        </div>
        <div
          class="overlay-0-2-39"
        />
        <div
          class="wrapper-0-2-41"
        >
          <div
            class="modal-0-2-42 modal-d0-0-2-48"
          >
            <button
              class="close-0-2-44"
              role="button"
            >
              x
            </button>
            <div
              class="content-0-2-45"
            />
          </div>
        </div>
        <div
          class="overlay-0-2-47"
        />
        <div
          class="wrapper-0-2-49"
        >
          <div
            class="modal-0-2-50 modal-d0-0-2-56"
          >
            <button
              class="close-0-2-52"
              role="button"
            >
              x
            </button>
            <div
              class="content-0-2-53"
            />
            <div
              class="footer-0-2-54"
            >
              Footer
            </div>
          </div>
        </div>
        <div
          class="overlay-0-2-55"
        />
        <div
          class="wrapper-0-2-57"
        >
          <div
            class="modal-0-2-58 modal-d0-0-2-64"
          >
            <button
              class="close-0-2-60"
              role="button"
            >
              x
            </button>
            <div
              class="content-0-2-61"
            />
          </div>
        </div>
        <div
          class="overlay-0-2-63"
        />
        <div
          class="wrapper-0-2-65"
        >
          <div
            class="modal-0-2-66 modal-d0-0-2-72"
          >
            <button
              class="close-0-2-68"
              role="button"
            >
              x
            </button>
            <div
              class="content-0-2-69"
            >
              Modal
            </div>
          </div>
        </div>
        <div
          class="overlay-0-2-71"
        />
        <div
          class="wrapper-0-2-73"
        >
          <div
            class="modal-0-2-74 modal-d0-0-2-80"
          >
            <div
              class="content-0-2-77"
            >
              Modal
            </div>
          </div>
        </div>
        <div
          class="overlay-0-2-79"
        />
      </body>
    }
  >
    <div
      className="wrapper-0-2-73"
      onClick={[Function]}
    >
      <div
        className="modal-0-2-74 modal-d0-0-2-80"
      >
        <div
          className="content-0-2-77"
        >
          Modal
        </div>
      </div>
    </div>
    <div
      className="overlay-0-2-79"
    />
  </Portal>
</Modal>
`;
